audit/options.go

```diff
@@ -158,3 +191,161 @@ func WithHeaderFormatter(f HeaderFormatter) Option {
+			return nil
+		}
+
+		var result []*exclusion
+
+		err := json.Unmarshal([]byte(e), &result)
+		if err != nil {
+			return fmt.Errorf("unable to parse exclusions: %w", err)
+		}
+
+		// Validate the exclusions
+		for _, exc := range result {
+			if err := exc.validate(); err != nil {
+				return err
+			}
+		}
+
+		o.withExclusions = result
+
+		return nil
+	}
+}
+
+// UnmarshalJSON handles unmarshalling JSON bytes (string representation) of a collection
+// of exclusion types into a Go type.
+func (e *exclusion) UnmarshalJSON(b []byte) error {
+	// Reference the JSON struct tags for exclusion.
+	const keyFields = "fields"
+	const keyCondition = "condition"
+
+	var err error
+
+	m := make(map[string]any)
+	if err = json.Unmarshal(b, &m); err != nil {
+		return err
+	}
+
+	// Parse fields
+	f, ok := m[keyFields]
+	if !ok {
+		return fmt.Errorf("exclusion '%s' missing", keyFields)
+	}
+	intermediateFields, ok := f.([]any)
+	if !ok {
+		return fmt.Errorf("unable to parse '%s': expected collection of %s; got: '%v'", keyFields, keyFields, f)
+	}
+	var fields []string
+	for _, v := range intermediateFields {
+		s := strings.TrimSpace(v.(string))
+		if s != "" {
+			fields = append(fields, s)
+		}
+	}
+	if len(fields) < 1 {
+		return fmt.Errorf("exclusion '%s' cannot be empty", keyFields)
+	}
+
+	// Set the fields now, so we can return early if we don't have an optional condition.
+	e.Fields = fields
+
+	// Optional condition
+	var eval *bexpr.Evaluator
+	c, ok := m[keyCondition]
+	if !ok {
+		// Return early as we've already set the exclusion.Fields
+		return nil
+	}
+
+	condition := strings.TrimSpace(fmt.Sprint(c))
+	if condition == "" {
+		// Return early as we've already set the exclusion.Fields
+		return nil
+	}
+
+	eval, err = bexpr.CreateEvaluator(condition)
+	if err != nil {
+		return fmt.Errorf("unable to parse expression '%s': %w", condition, err)
+	}
+
+	// Set the condition to the new evaluator.
+	e.Evaluator = eval
+
+	return nil
+}
+
+// validate attempts to parse the supplied fields to ensure they can be represented
+// as JSON pointers. When present, it will also evaluate the (optional) condition
+// using a sample RequestEntry and ResponseEntry.
+// NOTE: Validation will only be carried out against RequestEntry and ResponseEntry
+// types.
+func (e *exclusion) validate() error {
+	const op = "audit.(exclusion).validate"
+
+	if len(e.Fields) < 1 {
+		return fmt.Errorf("%s: exclusion doesn't contain any fields: %w", op, event.ErrInvalidParameter)
+	}
+
+	// Validate the 'fields' first (as the condition expression is optional)
+	for _, field := range e.Fields {
+		if _, err := pointerstructure.Parse(field); err != nil {
+			return fmt.Errorf("%s: unable to parse field '%s': %w", op, field, err)
+		}
+	}
+
+	// No condition expression for these fields, we can return early.
+	if e.Evaluator == nil {
+		return nil
+	}
+
+	// Generate a sample RequestEntry
+	req := new(RequestEntry)
+	if err := allocate.Zero(req); err != nil {
```

Comment:
TIL: `allocate.Zero`. Cool!

Reply:
I thought so too! We had to pull in the library for it, but after further work it looks like we aren't able to validate things fully in the same way we can for filtering (too many things that are flexible like `data`), so I had to pull it out 😢